app/dashboard/views.py

```diff
@@ -3442,7 +3443,145 @@ def hackathon_onboard(request, hackathon=''):
         'referer': referer,
         'is_registered': is_registered,
     }
-    return TemplateResponse(request, 'dashboard/hackathon_onboard.html', params)
+    return TemplateResponse(request, 'dashboard/hackathon/onboard.html', params)
+
+
+def hackathon_projects(request, hackathon=''):
+    q = request.GET.get('q', '')
+    order_by = request.GET.get('order_by', '-created_on')
+    filters = request.GET.get('filters', '')
+    print(q)
+
+    page = request.GET.get('page', 1)
+    try:
+        hackathon_event = HackathonEvent.objects.filter(slug__iexact=hackathon).latest('id')
+        profile = request.user.profile if request.user.is_authenticated and hasattr(request.user, 'profile') else None
```

Comment:
If the user is logged out, will this fail? I think `request.user` will be `None` so we may get an error like `Nonetype has no attribute "is_authenticated"` or something like that

Reply:
![image](https://user-images.githubusercontent.com/1358093/68248776-41113900-fffc-11e9-9c4c-10ae5ee7a4b1.png)
nop, is not failing if unlogged

Reply:
because `is_registered = HackathonRegistration.objects.filter(registrant=profile, hackathon=hackathon_event) if profile else None`  so if profile is  `None` also will be `is_registered`

Reply:
why not just force login ? `@login_required`

Reply:
because is not login_required, actually I will remove this part is not used anymore.